src/Common/src/System/Net/Security/NegotiateStreamPal.Unix.cs

```diff
@@ -98,13 +98,26 @@ internal static string QueryContextAuthenticationPackage(SafeDeleteContext secur
             ref SafeGssContextHandle context,
             SafeGssCredHandle credential,
             bool isNtlm,
+            SecurityBuffer cbt,
             SafeGssNameHandle targetName,
             Interop.NetSecurityNative.GssFlags inFlags,
             byte[] buffer,
             out byte[] outputBuffer,
             out uint outFlags,
             out int isNtlmUsed)
         {
+            // If a TLS channel binding token (cbt) is available then get the pointer
+            // to the application specific data.
+            IntPtr cbtAppData = IntPtr.Zero;
+            int cbtAppDataSize = 0;
+            if (cbt != null)
+            {
+                int appDataOffset = Marshal.SizeOf<SecChannelBindings>();
+                Debug.Assert(appDataOffset < cbt.size);
```

Comment:
Is cbt.size something that's coming in externally?  I'm wondering if there's any situation where we could get corrupt data such that in a debug build this assert would fail but in release we'd end up with a negative cbtAppDataSize, and whether that could cause any harm.

Reply:
The size is fixed.  The token is constructed here: 
https://github.com/dotnet/corefx/blob/master/src/System.Net.Security/src/System/Net/Security/Pal.Managed/EndpointChannelBindingToken.cs